miqa/core/rest/exceptions.py

```diff
@@ -0,0 +1,28 @@
+import logging
+import uuid
+
+from django.http.response import Http404
+from rest_framework import status
+from rest_framework.exceptions import APIException
+from rest_framework.response import Response
+from rest_framework.views import exception_handler
+
+logger = logging.getLogger(__name__)
+
+
+def custom_exception_handler(exc, context):
+    # Call REST framework's default exception handler first,
+    # to get the standard error response.
+    response = exception_handler(exc, context)
+
+    if not isinstance(exc, APIException) and not isinstance(exc, Http404):
```

Comment:
The addition of the `Http404` check here made me a bit nervous, so I looked at the base `exception_handler`, and this behavior is documented:

> Any unhandled exceptions may return `None`, which will cause a 500 error to be raised.

So all we need to do here is:

```suggestion
    if response is None:
```